### DISCUSSION TITLE
Can USB connection on host machine be used with an extension?

### DISCUSSION THREAD

**Discussion by phipsitn** [coder/code-server#2682] [Q&A]
Hello,

I am a coach for a lego robotics team and we use VScode to program our robot.  The problem we have is everyone has chromebooks and only the coaches have windows/mac laptops.  We use an extension called EV3Dev browser which allows VS Code to connect to the robot and download code to the robot.  Also you can ssh into the robot to test code.  If I was to start a code-server on one of these machines is there a trick to get the code-server to allow the usb connection to pass thru?

Thanks.

**Comment by oxy** [comment]
`code-server` should just have all the permissions of the user account that it is running on by default - if you run `code-server` on a device with the robot connected to it, and with sufficient permissions, you should just be able to use it as if it was local!

**Reply by phipsitn** [reply]
Thanks, I did find that starting it with brew services did not allow the usb to be seen.  When I started the raw executable and got the security checks to allow the node services, then it worked great.  I just need to script exposing the port and IP address and I should be all good.
